### PR TITLE
integrate test-results recent-overview as a widget into patient-chart

### PR DESCRIPTION
**Pull request by nmalyschkin**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -81,6 +81,18 @@ function setupOpenMRS() {
       }
     ],
     extensions: [
+      {
+        id: "test-results-widget",
+        slot: "test-results-widget",
```
Comment (FlorianRappl) [inline src/index.ts]: ID and Slot should be different.

The slot should be suffixed with `-slot`, or a given slot should be chosen. In this guess you want to have this part in the summary dashboard? Choose `patient-chart-summary-dashboard-slot` as slot.

File: src/widgets/test-results/overview/recent-overview.component.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
+import React from "react";
+
+import { Table16, ChartLine16 } from "@carbon/icons-react";
```
Comment (FlorianRappl) [inline src/widgets/test-results/overview/recent-overview.component.tsx]: Import from the respective submodules.

File: src/widgets/test-results/overview/recent-overview.component.tsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  Table,
+  TableContainer,
+  TableHead,
+  TableHeader,
+  TableRow,
+  TableCell,
+  TableBody,
+  TableToolbarContent,
+  TableToolbar
+} from "carbon-components-react";
```
Comment (FlorianRappl) [inline src/widgets/test-results/overview/recent-overview.component.tsx]: Import from the respective submodules.

File: src/widgets/test-results/overview/recent-overview.component.tsx [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+import { useHistory } from "react-router";
```
Comment (FlorianRappl) [inline src/widgets/test-results/overview/recent-overview.component.tsx]: `react-router-dom`

Comment (FlorianRappl) [reply src/widgets/test-results/overview/recent-overview.component.tsx]: (And yes, even though `react-router-dom` includes `react-router` note that in general this *may* make a difference. We only share `react-router-dom` (https://github.com/openmrs/openmrs-esm-core/blob/master/packages/esm-app-shell/src/dependencies.ts#L6) which means `react-router` would be bundled in and result in weird / wrong behavior...

**Review by FlorianRappl — COMMENTED** [review]
Great work! :beers:

File: src/index.ts (hunk 1 above)
Comment (nmalyschkin) [reply src/index.ts]: `patient-chart-summary-dashboard-slot` does not exist as a slot. All other widgets use a 1-to-1 mapping with their defined slots. Is this going to change in the near future?

Comment (FlorianRappl) [reply src/index.ts]: > patient-chart-summary-dashboard-slot does not exist as a slot.

If slots exist or don't exist does not make a difference. You also don't need to attach to a slot immediately.

> All other widgets use a 1-to-1 mapping with their defined slots.

This is / was just a temporary workaround to get them in as extensions.

> Is this going to change in the near future?

Yes - I think this was announced the last two weekly dev standups. We had the longer session on Monday (sorry during your vacation) where this system was proposed in greater depth. I think the video recording for it was shared already by Grace.

**Comment by FlorianRappl** [comment]
Cannot review when checks are failing. Are you committing with `--no-verify`? Please enable the hooks to get alarmed before pushing.

> /home/runner/work/openmrs-esm-patient-chart-widgets/openmrs-esm-patient-chart-widgets/src/widgets/test-results/overview/overview.component.tsx
> Error:   40:3  error  Unexpected console statement  no-console

**Comment by nmalyschkin** [comment]
> Cannot review when checks are failing. Are you committing with `--no-verify`? Please enable the hooks to get alarmed before pushing.

My bad, my `yarn link` was messing with the type checking, witch swallowed the failing check. It's fixed now.

**Review by FlorianRappl — APPROVED** [review]
LGTM - thanks :beers:!

**Pull request merged**